pkgs/_analyzer_macros/test/golden_test.dart

```diff
@@ -0,0 +1,81 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+import 'dart:isolate';
+
+import 'package:_analyzer_macros/macro_implementation.dart';
+import 'package:analyzer/dart/analysis/analysis_context.dart';
+import 'package:analyzer/dart/analysis/analysis_context_collection.dart';
+import 'package:analyzer/dart/analysis/results.dart';
+import 'package:analyzer/src/summary2/macro_injected_impl.dart' as injected;
+import 'package:test/test.dart';
+
+void main() {
+  late AnalysisContext analysisContext;
+
+  group('analyzer with injected macro impl query result matches golden', () {
+    setUp(() async {
+      // Set up analyzer.
+      final directory = Directory.current.absolute.uri
+          .resolve('../../introspection_goldens/foo');
+      final contextCollection =
+          AnalysisContextCollection(includedPaths: [directory.toFilePath()]);
+      analysisContext = contextCollection.contexts.first;
+      injected.macroImplementation = await AnalyzerMacroImplementation.start(
+          packageConfig: Isolate.packageConfigSync!);
+    });
+
+    for (final file in Directory.fromUri(
```

Comment:
nit: assign this directory logic to a top level variable, its a fair bit to repeat

Reply:
Done.